packages/api/src/hooks/p2p/useOrderDispute.ts

```diff
@@ -0,0 +1,78 @@
+import { useCallback, useMemo } from 'react';
+import useMutation from '../../useMutation';
+import useInvalidateQuery from '../../useInvalidateQuery';
+
+type TOrderDisputePayload = NonNullable<
+    Parameters<ReturnType<typeof useMutation<'p2p_order_dispute'>>['mutate']>
+>[0]['payload'];
+
+/** A custom hook that disputes a P2P order.
+ *
+ * To dispute an order, specify the following payload arguments in the `mutate` call:
+ * @example
+ *  mutate({
+ *      id: "1234",
+        dispute_reason: "seller_not_released",
+    });
+ *
+*/
+const useOrderDispute = () => {
+    const invalidate = useInvalidateQuery();
+    const {
+        data,
+        mutate: _mutate,
+        ...rest
+    } = useMutation('p2p_order_dispute', {
+        onSuccess: () => {
+            invalidate('p2p_order_info');
+        },
```

Comment:
Once an order has been disputed, an update could come immediately from `p2p_order_info` to update the current order information, thus this endpoint can be invalidated after an order has been disputed